src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,124 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_NODES_LENGTH = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private int loaded = (int) (DEFAULT_NODES_LENGTH * LOAD_FACTOR);
+    private int size;
+    private Node<K, V>[] nodes;
+
+    public MyHashMap() {
+        nodes = new Node[DEFAULT_NODES_LENGTH];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= loaded) {
+            resize();
+        }
+        int index = hash(key);
+        Node<K, V> currentNode = new Node<>(index, key, value);
+        if (nodes[index] == null) {
+            nodes[index] = currentNode;
+            size++;
+            return;
+        }
+        Node current = nodes[index];
+        while (current != null) {
+            if (current.getKey() == null && current.getKey() == key
```

Comment:
and everywhere - use fields, not getters, please

```suggestion
            if (current.key == key
```